compiler/rustc_hir_analysis/src/check/inherited.rs

```diff
@@ -60,7 +60,7 @@ pub struct Inherited<'tcx> {
     pub(super) deferred_asm_checks: RefCell<Vec<(&'tcx hir::InlineAsm<'tcx>, hir::HirId)>>,
 
     pub(super) deferred_generator_interiors:
-        RefCell<Vec<(hir::BodyId, Ty<'tcx>, hir::GeneratorKind)>>,
+        RefCell<Vec<(hir::HirId, hir::BodyId, ty::TyVid, hir::GeneratorKind)>>,
```

Comment:
It looks like all the users immediately call `mk_ty_var` on the stored `TyVid`. Could this be a `Ty` instead?